src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/String16FlyweightGenerator.java

```diff
@@ -235,14 +240,23 @@ private MethodSpec setStringMethod()
                     .returns(stringType.nestedClass("Builder"))
                     .addParameter(String.class, "value")
                     .addParameter(Charset.class, "charset")
+                    .beginControlFlow("if (value == null)")
+                    .addStatement("int newLimit = offset() + FIELD_SIZE_LENGTH")
+                    .addStatement("checkLimit(newLimit, maxLimit())")
+                    .addStatement("buffer().putShort(offset(), (short) -1)")
+                    .addStatement("limit(newLimit)")
+                    .nextControlFlow("else")
+                    .beginControlFlow("if (value.length() >= 65535)")
+                    .addStatement("throw new IllegalArgumentException("+
+                            "String.format(\"Value '%s' too big for String field (length < 65535)\", value))")
+                    .endControlFlow()
```

Comment:
Suggest generating `checkLength(int)` method with implementation:

```
private static void checkLength(
    int length)
{
    final int maxLength = 65534;
    if (length > maxLength)
    {
        final String msg = String.format("length=%d is beyond maximum length=%d", length, maxLength);
        throw new IndexOutOfBoundsException(msg);
    }
}
```

then call `checkLength(charBytes.length)` at line 254 below.

Note: there is a bug for multi-byte character UTF-8 strings with this current code, as `value.length()` is not the same as `charBytes.length` for multi-byte characters.